parser/Module.md

```diff
@@ -109,3 +109,111 @@ public function main() {
 ```
 
 *Note:* `parse` function returns `anydata` type when success, and it needs to be cast to the relevant FHIR Resource type.
+
+**03. Parse and validation with a valide FHIR data**
+
+In this approach the parser will attempt to parse and apply to the validation the given FHIR resource payload to given resource type profile.
+
+```ballerina
+import ballerina/log;
+import ballerinax/health.fhir.r4.parser;
+import ballerinax/health.fhir.r4.uscore501;
+public function main() {
+    json patientPayload = {
+        "resourceType": "Patient",
+        "id": "example-patient",
+        "text": {
+            "status": "generated",
+            "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">John Doe</div>"
+        },
+        "identifier": [
+            {
+                "system": "http://example.com/patient-ids",
+                "value": "12345"
+            }
+        ],
+        "extension": [
+            {
+                "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race",
+                "valueCodeableConcept": {
+                    "coding": [
+                        {
+                            "system": "http://hl7.org/fhir/v3/Race",
+                            "code": "2106-3",
+                            "display": "White"
+                        }
+                    ]
+                }
+            }
+        ],
+        "name": [
+            {
+                "use": "official",
+                "family": "Doe",
+                "given": [
+                    "John"
+                ]
+            }
+        ],
+        "gender": "male",
+        "birthDate": "2000-01-01"
+    };
+    do {
+        uscore501:USCorePatientProfile patientModel = check parser:parseWithValidation(patientPayload, uscore501:USCorePatientProfile).ensureType();
+        log:printInfo(string `Patient name : ${patientModel.name[0].toString()}`);
+    } on fail error parseError {
+        log:printError(string `Error occurred while parsing : ${parseError.message()}`, parseError);
+    }
+}
+```
+
+*Note:* `parseWithValidation` function returns `anydata` type when success, and it needs to be cast to the relevant FHIR Resource type. It will return validation error if it is fails.
+
+**04. Parse and validation with invalid FHIR data**
```

Comment:
Do we need a failure scenario in the module.md?
IMO Bal central documentation has to be easier to follow and having negative scenarios makes it bulky. WDYT?

Reply:
If you look from the validation perspective, a negative scenario is the main thing we need to test.

Reply:
+1